### ISSUE TITLE
I have some doubts about this usage, can someone give me an answer?

### ISSUE THREAD

**Issue by medition** [01org/TPM2.0-TSS#326]
`
        if( command != MS_SIM_CANCEL_ON && command != MS_SIM_CANCEL_OFF &&
                command != MS_SIM_POWER_ON && command != MS_SIM_POWER_OFF )
        {
            rval = StartCriticalSection( &tpmMutex, &functionString[0] );
            if( rval == TSS2_RC_SUCCESS )
            {
                criticalSectionEntered = 1;
            }
            else
            {
                goto retOtherCmdServer;
            }
        }
`

In function command server, why command MS_SIM_CANCEL_ON,MS_SIM_CANCEL_OFF,MS_SIM_POWER_ON,MS_SIM_POWER_OFF. if mutiple command come， it may be error？

**Comment by medition** [comment]
In resourcemgr.c, ResourceMgrReceiveTpmResponse function call TestForLoadedHandles(), what is the purpose of this function？

**Comment by medition** [comment]
In OtherCmdServer and TpmCmdServer，using select( serverStruct->connectSock+1, &readFds, 0, 0, 0 )， it only select for on socket fd，i read several articles,  not like this，they using select for mutiple socket fds。  And when i start 2000 client，it will make resourcemgr panic, when i remove select operation, resourcemgr work rightly.

**Comment by flihp** [comment]
ROFL yeah I Just responded to your #325 with an answer that is similar to what I'll write here: the resourcemgr is buggy and has been a real headache since I took over this project. When I first started trying to debug a few random crashes in it I set up a test similar to what you're doing (lots of parallel clients doing simple operations) and it completely melted down almost immediately. After looking at the threading code and all of the global data I just decided to start a reimplementation which is nearly done.

I'd much rather invest time getting my re-write out the door than fixing the existing resourcemgr code since I intend to delete it as soon as the replacementation is ready. So if you're willing to submit patches I'll review them but eventually this code will go away. I should probably put a note in the README about this.
